### ISSUE TITLE
[Accessibility] [Date Picker > Clearing the Date]: Focus does not return to the date input field after activating 'Clear date' button.

### ISSUE THREAD

**Issue by olehnoskov-cengage** [cengage/react-magma#2100]
[A11Y-559](https://jira.cengage.com/browse/A11Y-559)
[A11Y-588](https://jira.cengage.com/browse/A11Y-588)
[MTS-28375](https://jira.cengage.com/browse/MTS-28375)

**Issue 1:
[Accessibility] [Date Picker > Clearing the Date]: Focus does not return to the date input field after activating 'Clear date' button.**

Platform:
Windows 11 / Chrome

Steps to reproduce:
1. Open the URL: https://react-magma.cengage.com/version/4.9.1
2. Now Navigate to side navigation and expand 'Components' button.
3. Now activate 'Date picker' link under API
4. Now navigate to 'Clearing the date section and verify.

Actual Result:
The selected date is cleared from the input field, but keyboard focus is not returned to the editable date input field or any meaningful element.

Expected Result:
Once the 'Clear date' button is activated, focus should return to the date input field, allowing the user to re-enter or pick a new date without manually navigating back..

Instances:
1. [Accessibility][Form> Basic Usage]: Form displays validation errors, but focus remains on the Submit button, requiring users to manually search for the first error field.
https://app.screencast.com/ypNKwGcLzss5i

WCAG Guideline:
2.4.3 Focus Order Level A

User Impact:
Keyboard-only and screen reader users lose navigation continuity. This interrupts workflow and may require excessive tabbing or re-navigation to continue interacting with the form.

Recommendation:
After the 'Clear date' action is executed, programmatically move focus back to the corresponding date input field using JavaScript.

Screencast:
https://app.screencast.com/RmNJ6KVJf49ac

**Example problem. Component works as expected**

**Issue 2:
[Accessibility] [Global] [Datepicker] [Basic Usage] > Calendar widget help button has unclear accessible name; should announce "Calendar widget keyboard instructions".**

Environment:
Windows 11 / Chrome 137 / JAWS 2025
Windows 11 / Edge 136 / NVDA 2025
Mac 15 / Safari / VoiceOver

Steps to Reproduce:
1. Open the URL – https://react-magma.cengage.com/version/4.9.1/api/date-pickers/
2. Turn on a screen reader (JAWS/NVDA/VoiceOver).
3. Launch 'Datepicker' component page from left navigation.
4. On the 'Datepicker' page, under 'Basic Usage'.
5. Navigate to the calendar widget help button in the Datepicker calendar widget component.
6. Listen to the accessible name announced by the screen reader.

Actual Result:
The screen reader announces the button as "calendar widget help button". Here visually, the label indicates that it provides keyboard instructions, but this is not reflected in the accessible name.

Refer to Screencast: https://app.screencast.com/4O4itP2wmqnbP

Expected Result:
The screen reader should announce the button with visual label as "Keyboard instructions for Calendar widget" button.

WCAG Guideline:
2.5.3 Label in name (Level A)

User Impact:
Screen reader users may not understand that the button provides keyboard instructions, potentially missing important accessibility guidance for interacting with the calendar widget.

Bug Fix Recommendation:
Update the accessible name of the button using the following:

Use aria-label="Keyboard instructions for calendar widget" on the button.

**Issue 3:
 Screen reader not announcing the tolltip text which appear after hover on keyboard image inside calendar.**
Platform:
Windows 11 / Chrome /JAWS 2025
Edge/ NVDA 2025
Mac15/ Safari/ Voiceover

Steps to reproduce:
1. Open the URL "https://nglsync.cengage.com/rostering/Account/LogOn?DistrictLoginCode=JVXZ".
2. Click on the "Continue" button.
3. Login with the Teacher credentials(email & password).
4. Navigate near "Launch" for HS Physics and activate.
5.Select assignment from menu
6. Select a Material and on next page activate "Assign"
button.
7.Open the calendar which is associate with start/due date and observe that screen reader is not announcing about the tooltip content.

Actual Result:
For keyboard image icon inside the calendar have a tooltip with the text "Keyboard instruction" on hover. However, when the image receives keyboard focus, the screen reader announces it only as a “calender widget help button” instaed of tooltip text.

Expected Result:
On focus, screen reader should announce the tooltip text ("Keyboard instruction").
Tooltip content should be programmatically associated with the icon using ARIA.

Instances 1: Same issue is repro for "Assign Activity" from "Search".
Steps to reproduce:
1. Open the URL "https://nglsync.cengage.com/rostering/Account/LogOn?DistrictLoginCode=JVXZ".
2. Click on the "Continue" button.
3. Enter Teacher email and password and do sign in.
4. Select "My course" from main menu.
5. Select the course key: "MTPP9SQ54DWM" from HS Physics and launch the course from the "Launch course" button.
6. Assign the activity from Search>Results.

WCAG Guideline:
4.1.2 – Name, Role, Value

User Impact:
Screen reader and keyboard-only users do not receive important instructions for operating the calendar.
Leads to confusion, inaccessibility, and potentially task failure (e.g., unable to select a date).
Tooltip fails to serve its purpose for a major group of users.

Recommendation:
Use aria-describedby to associate the tooltip content with the icon, and ensures that on focus, screen readers will announce the tooltip content.

Screencast:
https://app.screencast.com/EteswYiwxRyVM
